blocks/social-share/social-share.js

```diff
@@ -31,7 +34,13 @@ export default async function decorate(block) {
   }
 
   if (window.location.hostname !== 'www.servicenow.com' && window.location.hostname !== 'localhost') {
-    block.dataset.url = `https://www.servicenow.com${window.location.pathname}.html`;
+    const pubDateStr = getMetadata('publication-date');
+    if (pubDateStr) {
+      const pubDate = new Date(pubDateStr);
+      if (pubDate < esdCutOff) {
+        block.dataset.url = `https://www.servicenow.com${window.location.pathname}.html`;
+      }
+    }
```

Comment:
otherwise we should simply use `https://www.servicenow.com${window.location.pathname}` without `.html`.